scripts/auth-common.sh

```diff
@@ -15,7 +13,38 @@ else
 fi
 
 IRONIC_HTPASSWD_FILE=/etc/ironic/htpasswd
+if [[ -f "/auth/ironic/username" ]]; then
+    read -r IRONIC_HTPASSWD_USERNAME<"/auth/ironic/username"
+fi
+IRONIC_HTPASSWD_USERNAME=${IRONIC_HTPASSWD_USERNAME:-}
+if [[ -f "/auth/ironic/password" ]]; then
+    read -r IRONIC_HTPASSWD_PASSWORD<"/auth/ironic/password"
+fi
+IRONIC_HTPASSWD_PASSWORD=${IRONIC_HTPASSWD_PASSWORD:-}
+if [[ -n "${IRONIC_HTPASSWD_USERNAME}" ]]; then
+    IRONIC_HTPASSWD="$(htpasswd -n -b -B "${IRONIC_HTPASSWD_USERNAME}" "${IRONIC_HTPASSWD_PASSWORD}")"
+fi
+IRONIC_HTPASSWD=${IRONIC_HTPASSWD:-${HTTP_BASIC_HTPASSWD:-}}
+if [[ -n "${IRONIC_HTPASSWD}" ]]; then
+    export IRONIC_HTPASSWD
+fi
+
 INSPECTOR_HTPASSWD_FILE=/etc/ironic-inspector/htpasswd
+if [[ -f "/auth/inspector/username" ]]; then
```

Comment:
A bit of code duplicate here, but since inspector is going away, I don't insist on refactoring.

Reply:
Yeah, I figured its going away soon so it should be okay hopefully.